.github/workflows/nestbuildmatrix.yml

```diff
@@ -2,32 +2,444 @@
+        with:
+          fetch-depth: 0
+
+      - name: "Set up Python 3.x"
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: "Install dependencies"
+        run: |
+          pip install pydocstyle
+
+      - name: "Check for unused nest::names..."
+        run: |
+          python build_support/check_unused_names.py
+
+  forbidden_types:
+    runs-on: "ubuntu-20.04"
+    steps:
+      - name: "Checkout repository content"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: "Check for forbidden types..."
+        run: |
+          ./build_support/check_forbidden_types.sh
+
+  lychee:
+    runs-on: "ubuntu-20.04"
+    steps:
+      - name: "Checkout repository content"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+        
+      - name: "Install dependencies"
+        run: |
+          #sudo apt install python3-docutils
+          wget --progress=dot:mega 'https://github.com/lycheeverse/lychee/releases/download/v0.8.2/lychee-v0.8.2-x86_64-unknown-linux-gnu.tar.gz'
+          echo '12c27c9b6d551aea9178080d27de75ed26125b130722153399259432f723e606  lychee-v0.8.2-x86_64-unknown-linux-gnu.tar.gz' | sha256sum -c
+          tar -xzf 'lychee-v0.8.2-x86_64-unknown-linux-gnu.tar.gz'
+
+      - name: "Run lychee..."
+        continue-on-error: true
+        run: "pwd; ls -l; ./lychee build_support/ doc/{*.txt,*.md,htmldoc} examples/ lib* models/ nest* pynest/ testsuite/ *.md"
+        # TODO: doc/auto_examples and doc/models need to be added here, but exists only after a `make html`
+        env:
+          # Required, set by GitHub actions automatically:
+          # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret
+          # to overcome rate limiting
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  pydocstyle:
+    runs-on: "ubuntu-20.04"
+    steps:
+      - name: "Checkout repository content"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: "Set up Python 3.x"
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: "Install dependencies"
+        run: |
+          pip install pydocstyle
+
+      - name: "Run pydocstyle..."
+        run: |
+          pydocstyle pynest/
+
+  mypy:
+    runs-on: "ubuntu-20.04"
+    #needs: [pydocstyle, pycodestyle, pylint, flake8]
+    steps:
+      - name: "Checkout repository content"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: "Set up Python 3.x"
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: "Install Linux system dependencies"
+        run: |
+          pip install mypy matplotlib numpy scipy data-science-types
+
+      - name: "Run mypy..."
+        run: |
+          mypy pynest/
+
+  pytest-linux:
```

Comment:
Since we are still working on porting the SLI tests to python, we can turn this job off until then.